static/js/fit.js

```diff
@@ -981,24 +981,11 @@ $( document ).ready(function() {
       }
   });
 
-  delegate('#text-segment', 'submit', (e,t) => {
-    let mysave = $('#fit_wysiwyg_editor').html();
-    $('#text_segment_content').val(mysave);
-  });
-
-  delegate('#lesson-edit-form', 'submit', (e,t) => {
-    let mysave = $('#fit_wysiwyg_editor').html();
-    $('#further_reading').val(mysave);
-  });
-
-  delegate('#course-edit-form', 'submit', (e,t) => {
-    let mysave = $('#fit_wysiwyg_editor').html();
-    $('#course_summary').val(mysave);
-  });
-
-  delegate('#add-edit-resource', 'submit', (e,t) => {
-    let mysave = $('#fit_wysiwyg_resource').html();
-    $('#resource_description').val(mysave);
+  delegate('[data-fit-wysiwyg]', 'submit', (e,t) => {
+    let wysiwygId = t.dataset.fitWysiwygId;
+    let mysave = $(`[data-fit-wysiwyg-div-${wysiwygId}]`).html();
+    let testarea = $(`[data-textarea-wysiwyg-${wysiwygId}]`);
+    testarea.val(mysave);
```

Comment:
Very nice work, @mkarpicz!

You can put the ID directly into the `data-fit-wysiwyg` property to define your target element and not need the `data-fit-wysiwyg-id` attribute.  This is how a lot of Bootstrap stuff works. :)

Reply:
The dynamic attributes are a bit strange, why not try using two data attributes to specify the preview and the input target?

Reply:
Or another way to do it...

```.html
<form data-fit-wysiwyg>
    <textarea name="course_summary" style="display: none" data-textarea-wysiwyg-input></textarea>
    <div data-fit-wysiwyg-preview></div>
</form>
```

```.js
delegate('[data-fit-wysiwyg]', 'submit', (e, t) => {
    let p = t.closest('[data-fit-wysiwyg]');
    let mysave = p.querySelector('[data-fit-wysiwyg-input]').html();
    let testarea = p.querySelector('[data-fit-wysiwyg-preview]').html();
    /* ... */
});
```